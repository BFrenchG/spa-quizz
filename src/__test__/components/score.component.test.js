// @flow

import React from 'react';
import { shallow } from 'enzyme';
import Score from "../../components/score.component";



const setup = (setupProps = {}) => {

    const defaultProps = {
        score: 75,
        retry: jest.fn()
    };

    const props = {...defaultProps, ...setupProps};

    const wrapper = shallow(<Score score={props.score} retry={props.retry} />);

    return {
        wrapper
    };
};

describe('Score', () => {
    test('renders without crashing', () => {
        const { wrapper } = setup();
        expect(wrapper).toMatchSnapshot();
    });
});