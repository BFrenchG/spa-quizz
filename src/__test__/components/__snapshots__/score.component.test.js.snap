// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Score
    retry={[MockFunction]}
    score={75}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <p>
          Your score is: 
          <b>
            75
            %
          </b>
        </p>,
        <WithStyles(Button)
          color="secondary"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": "2px",
            }
          }
          variant="raised"
        >
          Retry
        </WithStyles(Button)>,
      ],
      "raised": false,
      "style": Object {
        "margin": "5px 0",
        "padding": "5px 15px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Your score is: ",
            <b>
              75
              %
            </b>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Your score is: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                75,
                "%",
              ],
            },
            "ref": null,
            "rendered": Array [
              75,
              "%",
            ],
            "type": "b",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Retry",
          "color": "secondary",
          "onClick": [Function],
          "style": Object {
            "marginLeft": "2px",
          },
          "variant": "raised",
        },
        "ref": null,
        "rendered": "Retry",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <p>
            Your score is: 
            <b>
              75
              %
            </b>
          </p>,
          <WithStyles(Button)
            color="secondary"
            onClick={[Function]}
            style={
              Object {
                "marginLeft": "2px",
              }
            }
            variant="raised"
          >
            Retry
          </WithStyles(Button)>,
        ],
        "raised": false,
        "style": Object {
          "margin": "5px 0",
          "padding": "5px 15px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Your score is: ",
              <b>
                75
                %
              </b>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Your score is: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  75,
                  "%",
                ],
              },
              "ref": null,
              "rendered": Array [
                75,
                "%",
              ],
              "type": "b",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Retry",
            "color": "secondary",
            "onClick": [Function],
            "style": Object {
              "marginLeft": "2px",
            },
            "variant": "raised",
          },
          "ref": null,
          "rendered": "Retry",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
